### ISSUE TITLE
Geolocate authors to countries based on affiliations

### ISSUE THREAD

**Issue by dhimmel** [greenelab/iscb-diversity#8]
We'd like to investigate whether we can use affiliation information to assign countries to authors.

Here is a code snippet I created in the past:

```python
geolocator = geopy.geocoders.Nominatim(user_agent='test')
address = '1G Royal Parade, Parkville, Victoria 3052, Australia'
location = geolocator.geocode(address, addressdetails=True)
location.raw['address']['country_code']
```

**Comment by dhimmel** [comment]
I completed a first pass at affiliation to country conversion in [`09.affiliations-to-countries.ipynb`](https://github.com/greenelab/iscb-diversity/blob/605137b226cb2b1fc860ee78c3869419792535cb/09.affiliations-to-countries.ipynb). Check out the [raw data here](https://raw.githubusercontent.com/greenelab/iscb-diversity/605137b226cb2b1fc860ee78c3869419792535cb/data/affiliations/geocode.jsonl).

Here is what a line looks like:

```json
{"affiliation": "Bioinformatics Division, The Walter and Eliza Hall Institute of Medical Research, Parkville, Victoria, Australia. mritchie@wehi.edu.au", "country_geocode": "AU", "countries_geotext": {"AU": 1, "US": 1, "CA": 1}, "countries": ["AU"]}
```

countries_geotext comes from the geotext python package which looks for predefined place names.

country_geocode comes from OpenStreetMap nomatim and is the country containing the top result.

countries is the consensus countries in the address computed using an algorithm I developed that takes countries_geotext and country_geocode as input.
